Mathlib/LinearAlgebra/Basis/Flag.lean

```diff
@@ -0,0 +1,109 @@
+/-
+Copyright (c) 2023 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov, Patrick Massot
+-/
+import Mathlib.LinearAlgebra.Dual
+import Mathlib.Data.Fin.FlagRange
+
+/-!
+# Flag of submodules defined by a basis
+
+In this file we define `Basis.flag b k`, where `b : Basis (Fin n) R M`, `k : Fin (n + 1)`,
+to be the subspace spanned by the first `k` vectors of the basis `b`.
+
+We also prove some lemmas about this definition.
+-/
+
+open Set Submodule
+
+namespace Basis
+
+section Semiring
+
+variable {R M : Type _} [Semiring R] [AddCommMonoid M] [Module R M] {n : ℕ}
+
+/-- The subspace spanned by the first `k` vectors of the basis `b`. -/
+def flag (b : Basis (Fin n) R M) (k : Fin (n + 1)) : Submodule R M :=
+  .span R <| b '' {i | i.castSucc < k}
```

Comment:
It looks like the TODO of `partialSups` would allow this to be spelt `partialSups (fun i => R ∙ b i.castSucc)`. @YaelDillies, is that TODO easy?

Reply:
Do you think this PR has to wait for that TODO, or should we merge it now and update the definition later?

Reply:
The TODO is pretty easy, yeah. All we need is a recursor in terms of `order.succ`.

Reply:
Good point. I like that.

Reply:
IMHO, we can have a lemma that relates `Basis.flag` to `partialSups` but this shouldn't be the definition. Instead, we may want to add `Basis.submodule (s : Set ι) : Submodule R M := Submodule.span R (b '' s)` (I saw that the Brouwer Fixed Point project uses this).

Reply:
I moved supporting lemmas to new PRs so that we can merge them while discussing this PR.

Reply:
I agree with @urkud here.